### DISCUSSION TITLE
Handling SIGTERM gracefully with the app router

### DISCUSSION THREAD

**Discussion by osdiab** [vercel/next.js#51402] [Help]
### Summary

[The docs for NextJS using the App Router for `Manual graceful shutdowns`](https://nextjs.org/docs/app/building-your-application/deploying#manual-graceful-shutdowns) use the `pages/document.js` file to handle shutdown signals, though my project doesn't even have a `pages/` directory anymore now that everything is moved over to App Router. Is it still the "proper" way to handle this, or is there some other App Router-native method of handling shutdown signals? Thanks!

**Comment by RileyIsland** [comment]
I am still in local development phase on a personal project without a deployment path yet, but reading up on this currently as I go through the full documentation for next.js.

@osdiab did you try placing the code block in `app/layout.tsx` or wherever your root layout(s) is(are)?

**Reply by osdiab** [reply]
Didn’t try, I ended up using Vercel instead of hosting on a separate cloud provider so I stopped experiencing the errors that I was running into
